Poof/Assets/Scripts/Shadow/Shadow.cs

```diff
@@ -0,0 +1,52 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+using System;
+
+public class Shadow : MonoBehaviour
+{
+    [SerializeField, SerializeReference]
+    public ShadowTarget target = null;
+
+    int id = -1;
+    public int ID
+    {
+        get
+        {
+            if (id == -1)
+                id = ShadowManager.idCounter++;
+            return id;
+        }
+    }
+
+    [SerializeField]
+    int TIME_OFFSET = 1500;
```

Comment:
Shadow should always be visible and may swap to other side once player is half way threw. ...unless it looks weird! May change with jump movement and animations. "Time_offset" at 500 is to slow 120 left nice.  I like having the shadow go farther away from the player the faster they are going.

Reply:
breaking down your comments
1. Shadow should always be visible and may swap to other side once player is half way threw. ...unless it looks weird! May change with jump movement and animations. 
2. "Time_offset" at 500 is to slow 120 left nice. 
3. I like having the shadow go farther away from the player the faster they are going.


1. so the shadow gets alot more complicated if we dont want it to exactly follow the player, at minimum i expect that it would end up clipping through alot of stuff. in the shadows current state once animations are added/art then the shadow should end up being out of sync with the player so it shouldn't ever end up hidden inside the player
2. ill commit that change
3. it already does that?